libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/impl/NotificationServiceFactoryImpl.java

```diff
@@ -26,6 +26,6 @@
 
     @Override
     public NotificationService create() {
-        return new EmailNotificationService();
+        return new EmailNotificationService(new AWSEmailProvider());
```

Comment:
Instead of hardcoding the AWSEmailProvider here, it should come from properties, so that it can be replaced with another EmailProvider for ZMS vs ZTS

Reply:
Yes, I've considered it and it is actually part of the original design. But that means there would be 2 separate factory types. One for NotificationService - currently only email but in the future could be something else like push messages to phone - and another for providers such as AWS SES, or some SMS messages provider.
The providers and types of Notification Services are coupled as there is no reason to Instantiate a NotificationService of one type (for example, EmailNotificationService) with a provider from another type (for example, some push notifications Provider).
So I've decided on using a single factory. Implementors can implement other factories that create EmailNotificationService with a different provider of their own or create a different type of NotificationService all together with its own providers.